flocker/provision/_digitalocean.py

```diff
@@ -0,0 +1,224 @@
+# ('9836874', u'522.4.0 (stable)', {'distribution': u'CoreOS'})
+# ('6370882', u'20 x64', {'distribution': u'Fedora'})
+# ('6370968', u'19 x64', {'distribution': u'Fedora'})
+# ('6372108', u'6.5 x64', {'distribution': u'CentOS'})
+# ('6372321', u'5.10 x64', {'distribution': u'CentOS'})
+# ('6372526', u'7.0 x64', {'distribution': u'Debian'})
+# ('6372581', u'6.0 x64', {'distribution': u'Debian'})
+# ('6374124', u'10.04 x64', {'distribution': u'Ubuntu'})
+# ('6374128', u'12.04.5 x64', {'distribution': u'Ubuntu'})
+# ('7053293', u'7.0 x64', {'distribution': u'CentOS'})
+# ('9801950', u'14.04 x64', {'distribution': u'Ubuntu'})
+# ('9801954', u'14.10 x64', {'distribution': u'Ubuntu'})
+IMAGE_NAMES = {
+    # It'd be better to use image ID here, but the following code is currently
+    # written to lookup image names...which would normally be good for
+    # readability but which in the case DigitalOcean are pretty meaningless.
+    'fedora-20': '20 x64',
+}
+
+
+def location_by_slug(driver, location_slug):
+    """
+    Look up a DigitalOcean by its short human readable "slug" code.
+
+    # XXX: ``libcloud.DigitalOceanDriver.list_locations`` discards the slug
+    # so we make a direct call to the v1 API and parse the returned dictionary.
+    # See https://issues.apache.org/jira/browse/LIBCLOUD-653
+
+    :param driver: The libcloud driver to query for sizes.
+    :param bytes location_slug: A DigitalOcean location "slug".
+    :returns: ``NodeLocation``.
+    """
+    result = driver.connection.request('/regions')
+    for location_dict in result.object['regions']:
+        if location_dict['slug'] == location_slug:
+            break
+    else:
+        raise ValueError("Unknown location slug.", location_slug)
+
+    return driver._to_location(location_dict)
+
+
+def get_ssh_key_id(driver, ssh_key_name):
+    """
+    """
+    # There's no high level API for this in the DigitalOcean driver.
+    response = driver.connection.request('/ssh_keys')
+    keys = response.object['ssh_keys']
+    try:
+        return [k['id'] for k in keys if k['name'] == ssh_key_name][0]
+    except IndexError:
+        raise ValueError("Unknown SSH keyname.", ssh_key_name)
+
+
+def digitalocean_provisioner(client_id, api_key, token, location, keyname):
+    """
+    Create a LibCloudProvisioner for provisioning nodes on DigitalOcean.
+
+    :param bytes client_id: A V1 API client ID.
+    :param bytes api_key: A V1 API key.
+    :param bytes token: A V2 API token.
+    :param bytes location: The slug for the location in which new nodes will be
+        created.
+    :param bytes keyname: The name of an existing ssh public key configured in
+       DigitalOcean. The provision step assumes the corresponding private key
+       is available from an agent.
+    """
+    # Import these here, so that this can be imported without
+    # installing libcloud.
+    from libcloud.compute.providers import get_driver, Provider
```

Comment:
I may be getting mixed up, but I've seen similar done elsewhere in Flocker with:

```
try:
    from libcloud.compute.providers import get_driver, Provider
    LIBCLOUD_INSTALLED = True
except:
    LIBCLOUD_INSTALLED = False
```

If these have the same consequences, let's try to be consistent. If not, please tell me the differences.